components/org.wso2.carbon.identity.api.server.claim.management/org.wso2.carbon.identity.rest.api.server.claim.management.v1/src/main/java/org/wso2/carbon/identity/rest/api/server/claim/management/v1/core/ServerClaimManagementService.java

```diff
@@ -1531,4 +1544,57 @@ private void validateClaimModificationEligibility() throws ClaimMetadataClientEx
+
+        try {
+            String organizationId =
+                    getOrganizationManager().resolveOrganizationId(ContextLoader.getTenantDomainFromContext());
+            return !getOrganizationManager().isPrimaryOrganization(organizationId);
+        } catch (OrganizationManagementException e) {
+            throw new ClaimMetadataClientException(Constant.ErrorMessage.ERROR_CODE_ERROR_RESOLVING_ORGANIZATION.
+                    getCode(), Constant.ErrorMessage.ERROR_CODE_ERROR_RESOLVING_ORGANIZATION.getDescription());
+        }
+    }
+
+    private void validateAttributeMappingUpdate(String claimID, LocalClaim incomingLocalClaim)
+            throws ClaimMetadataException {
+
+        Optional<LocalClaim>
+                existingLocalClaim = getClaimMetadataManagementService().getLocalClaim(base64DecodeId(claimID),
+                ContextLoader.getTenantDomainFromContext());
+
+        if (!existingLocalClaim.isPresent()) {
+            throw handleClaimManagementClientError(ERROR_CODE_LOCAL_CLAIM_NOT_FOUND, BAD_REQUEST, claimID);
+        }
+
+        populateDefaultProperties(existingLocalClaim.get());
+        if (Objects.hash(existingLocalClaim.get().getClaimProperties().entrySet()) !=
+                Objects.hash(incomingLocalClaim.getClaimProperties().entrySet())) {
+            throw handleClaimManagementClientError(ERROR_CODE_UNAUTHORIZED_ORG_FOR_CLAIM_PROPERTY_UPDATE, FORBIDDEN);
+        }
+
+        for (AttributeMapping existingMapping : existingLocalClaim.get().getMappedAttributes()) {
+            if (IdentityUtil.getPrimaryDomainName().equals(existingMapping.getUserStoreDomain())) {
+                Optional<AttributeMapping> incomingAttributeMapping = incomingLocalClaim.getMappedAttributes().stream()
+                        .filter(mapping -> IdentityUtil.getPrimaryDomainName().equals(mapping.getUserStoreDomain()))
+                        .findFirst();
+                // Not allowing to update the primary userstore attribute mapping for sub-orgs.
+                if (incomingAttributeMapping.isPresent() && !StringUtils.equals(existingMapping.getAttributeName(),
+                        incomingAttributeMapping.get().getAttributeName())) {
+                    throw handleClaimManagementClientError(ERROR_CODE_UNAUTHORIZED_ORG_FOR_ATTRIBUTE_MAPPING_UPDATE,
+                            FORBIDDEN, existingMapping.getUserStoreDomain());
+                }
+            }
+        }
+    }
+
+    private void populateDefaultProperties(LocalClaim localClaim) {
+
+        localClaim.getClaimProperties().putIfAbsent(PROP_DESCRIPTION, StringUtils.EMPTY);
+        localClaim.getClaimProperties().putIfAbsent(PROP_DISPLAY_ORDER, "0");
+        localClaim.getClaimProperties().putIfAbsent(PROP_READ_ONLY, FALSE);
+        localClaim.getClaimProperties().putIfAbsent(PROP_REQUIRED, FALSE);
+        localClaim.getClaimProperties().putIfAbsent(PROP_SUPPORTED_BY_DEFAULT, FALSE);
+    }
```

Comment:
What if new claim properties were introduced later? Seems unless the default value cases are not hanlded in this place, the sub org claim update will be failed. 
Need to take necessary actions to reduce maintainability cost

Reply:
Updated with [7bcd42b](https://github.com/wso2/identity-api-server/pull/771/commits/7bcd42b531f32433a8253deb53cd32e381ad7d5a) as discussed offline